### PR TITLE
Implement the API 

### PR DESCRIPTION
**Pull request by cosletmihai**

### CODE REVIEW

**Comment by cosletmihai** [comment]
Need help how to change the internal port of `fuseki` and how to extend the timeout for `gunicorn` (for fingerprinting larger files), as the connection closes before the fingerprinting execution is finished.

File: fingerprinter/adapters/sparql_adapter.py [hunk 1]
```diff
@@ -83,7 +83,8 @@ def create_dataset(self, dataset_name: str):
         }
 
         response = self.http_client.post(urljoin(self.triplestore_service_url, f"/$/datasets"),
-                                         auth=HTTPBasicAuth(config.FUSEKI_USERNAME, config.FUSEKI_PASSWORD),
+                                         auth=HTTPBasicAuth(config.RDF_FINGERPRINTER_FUSEKI_USERNAME,
```
Comment (mclaurentiu) [inline fingerprinter/adapters/sparql_adapter.py:86]: just as a remark, we should use HTTPS as much as possible, if we use basic auth

**Review by mclaurentiu — APPROVED** [review]

**Review by costezki — APPROVED** [review]
looks good

File: tests/unit/test_entrypoints_api_handlers.py [hunk 2]
```diff
@@ -0,0 +1,92 @@
...
+"""
+from io import BytesIO
+from unittest.mock import patch
+
+import pytest
+from werkzeug.datastructures import FileStorage
+
+
+@patch('fingerprinter.entrypoints.api.handlers.service_fingerprint_sparql_endpoint')
+def test_fingerprint_sparql_endpoint_success(mock_service_fingerprint_sparql_endpoint, api_client, tmpdir):
```
[the remark this answers is not in the document]
Comment (cosletmihai) [reply tests/unit/test_entrypoints_api_handlers.py:19]: it's not the mock that gets tested, but if the view behaves the way it is expected to behave: 
- get a sparql endpoint to fingerprint
- call the `service_fingerprint_sparql_endpoint`
- return a report

**Pull request merged**